### PR TITLE
[Ameba] Fix runtime hardfault error

### PR DESCRIPTION
**Pull request by pankore**
#### Problem
* Fix runtime hardfault error

#### Change overview
* Revise chipinterface.cpp files for all-clusters-app, lighting-app and ota-requestor-app

#### Testing
* Tested all-clusters-app, lighting-app and ota-requestor-app

### CODE REVIEW

File: examples/all-clusters-app/ameba/main/chipinterface.cpp [hunk 1]
```diff
@@ -157,23 +177,9 @@ extern "C" void ChipTest(void)
...
-    // Initialize device attestation config
-    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
-    NetWorkCommissioningInstInit();
-
-    if (RTW_SUCCESS != wifi_is_connected_to_ap())
-    {
-        // QR code will be used with CHIP Tool
-        PrintOnboardingCodes(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
-    }
+    chip::DeviceLayer::PlatformMgr().ScheduleWork(InitServer, reinterpret_cast<intptr_t>(nullptr));
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/ameba/main/chipinterface.cpp:180]: Why not just pass 0 for the int?

File: examples/ota-requestor-app/ameba/main/chipinterface.cpp [hunk 2]
```diff
@@ -136,18 +146,7 @@ extern "C" void ChipTest(void)
...
     }
 
-    chip::Server::GetInstance().Init();
-
-    // Initialize device attestation config
-    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
-    NetWorkCommissioningInstInit();
+    chip::DeviceLayer::PlatformMgr().ScheduleWork(InitServer, reinterpret_cast<intptr_t>(nullptr));
 
     InitOTARequestor();
```
Comment (bzbarsky-apple) [inline examples/ota-requestor-app/ameba/main/chipinterface.cpp:151]: Shouldn't InitOTARequestor be happening in InitServer?

File: examples/ota-requestor-app/ameba/main/chipinterface.cpp [hunk 3]
```diff
@@ -136,18 +146,7 @@ extern "C" void ChipTest(void)
...
-
-    while (true)
-        vTaskDelay(pdMS_TO_TICKS(50));
```
Comment (bzbarsky-apple) [inline examples/ota-requestor-app/ameba/main/chipinterface.cpp:148]: Why is this while loop being removed?

File: examples/ota-requestor-app/ameba/main/chipinterface.cpp (hunk 2 above)
Comment (pankore) [reply examples/ota-requestor-app/ameba/main/chipinterface.cpp]: This has been fixed.

File: examples/all-clusters-app/ameba/main/chipinterface.cpp (hunk 1 above)
Comment (pankore) [reply examples/all-clusters-app/ameba/main/chipinterface.cpp]: This has been fixed.

File: examples/ota-requestor-app/ameba/main/chipinterface.cpp (hunk 3 above)
Comment (pankore) [reply examples/ota-requestor-app/ameba/main/chipinterface.cpp:148]: This while loop is redundant.

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Approved, but note discussion in https://github.com/project-chip/connectedhomeip/issues/15295 where right now doing Server::Init from inside the Matter event loop will miss the StartUp event.  I am hoping we will fix that in a way that still allows the init to happen from the Matter event loop....

**Pull request merged**